src/main/java/me/minidigger/hangar/controller/api/AuthenticateApiController.java

```diff
@@ -24,18 +23,21 @@ public AuthenticateApiController(AuthenticationService service) {
 
     @Override
     public ResponseEntity<ApiSessionResponse> authenticate(SessionProperties body) {
-        if (body != null && body.isFake() != null && body.isFake()) {
+        if (body != null && body.isFake()) {
             return ResponseEntity.ok(service.authenticateDev());
         } else {
-            Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-            if (authentication instanceof HangarAuthentication) {
-                return ResponseEntity.ok(service.authenticateKeyPublic(body, ((HangarAuthentication) authentication).getUserId()));
-            } else if (authentication.getPrincipal().equals("anonymousUser")) {
-                return ResponseEntity.ok(service.authenticatePublic());
-            } else {
-                throw AuthUtils.unAuth();
-            }
+            return ResponseEntity.ok(service.authenticateKeyPublic(body == null ? null : body.getExpiresIn()));
         }
+            // TODO not sure if all this is needed because you don't need the hangarauth user id, all that is from the api key
```

Comment:
I think I created this cause I had issue with ano login, if authenticatePublic doesnt use the id, just remoce all of this